server/src/main/java/org/opensearch/search/startree/filter/provider/DimensionFilterMapper.java

```diff
@@ -235,6 +280,29 @@ public Comparator<Long> comparator() {
         return DimensionDataType.UNSIGNED_LONG::compare;
     }
 
+    @Override
+    public int compareValues(Object v1, Object v2) {
+        if (!(v1 instanceof Long) || !(v2 instanceof Long)) {
+            throw new IllegalArgumentException("Expected Long values for unsigned comparison");
+        }
+        return Long.compareUnsigned((Long) v1, (Long) v2);
+    }
+
+    @Override
+    public boolean isValueInRange(Object value, Object low, Object high, boolean includeLow, boolean includeHigh) {
+        long v = (Long) value;
+        long l = low != null ? (Long) low : 0L;
+        long h = high != null ? (Long) high : -1L; // -1L is max unsigned
```

Comment:
I didn't know that `-1L` is max unsigned.

Reply:
-1L has all bits set to 1 and when this bit pattern is interpreted as an unsigned long - it represents maximum possible value